providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -380,9 +381,8 @@ def create_pod(self, pod: V1Pod) -> V1Pod:
 
     async def watch_pod_events(self, pod: V1Pod, check_interval: int = 1) -> None:
         """Read pod events and writes into log."""
-        self.keep_watching_for_events = True
         num_events = 0
-        while self.keep_watching_for_events:
+        while not self.stop_watching_events:
             events = self.read_pod_events(pod)
```

Comment:
@jedcunningham you might find this interesting. 

there was actually a race condition here.

i found in the system tests that if await_pod_start ran first, then watch_pod_events would _set keep_watching_for_events back to true_ and nothing would ever set it false.

so these subtle changes ensure that once it is set to stop, it will stay that way.

Reply:
`not stop` -- do we need to change this do a double negative?